internal/transport/transport_test.go

```diff
@@ -2243,3 +2244,27 @@ func TestTCPUserTimeout(t *testing.T) {
 		}
 	}
 }
+
+// TestRequestInfoFoundInStreamContext tests whether data passed in as callheader data gets
+// propagated into the context object as a credentials.RequestInfo object.
+func TestRequestInfoFoundInStreamContext(t *testing.T) {
+	serverConfig := &ServerConfig{}
+	server, client, cancel := setUpWithOptions(t, 0, serverConfig, suspended, ConnectOptions{})
+	defer cancel()
+	defer server.stop()
+	defer client.Close()
+	ch := &CallHdr{
+		Method: "someService/Method",
+	}
+	s, err := client.NewStream(context.Background(), ch)
+	if err != nil {
+		t.Fatalf("client.NewStream() failed: %v", err)
+	}
+	ri, ok := s.ctx.Value(credentials.RequestInfoKey{}).(credentials.RequestInfo)
```

Comment:
This confirms that the stream's context, internal to the transport, has the `RequestInfo`, but not that it is available to the credentials.

I'd recommend adding something to `test/end2end_test.go`.  Use a simple test as a template (e.g. `TestGRPCMethod`), and implement a simple `TransportCredentials` that verifies the method name but otherwise does nothing.